### ISSUE TITLE
Windows --cuda_ext build fails due to missing canUse32BitIndexMath

### ISSUE THREAD

**Issue by gcp** [NVIDIA/apex#96]
Latest MSVC 2017 update, CUDA 10.0.130, PyTorch 1.0 release with python 3.6, apex from master branch.

```
   Creating library build\temp.win-amd64-3.6\Release\apex/optimizers/csrc\fused_adam_cuda.cp36-win_amd64.lib and object build\temp.win-amd64-3.6\Release\apex/optimizers/csrc\fused_adam_cuda.cp36-win_amd64.exp
fused_adam_cuda_kernel.obj : error LNK2001: unresolved external symbol "bool __cdecl at::cuda::detail::canUse32BitIndexMath(class at::Tensor const &,__int64)" (?canUse32BitIndexMath@detail@cuda@at@@YA_NAEBVTensor@3@_J@Z)
build\lib.win-amd64-3.6\fused_adam_cuda.cp36-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120
```

**Comment by gcp** [comment]
Some looking around says that function is supposed to be in caffe2_gpu.lib, but that is indeed on the link line:

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\DeepSjeng\train\venv10\lib\site-packages\torch\lib "/LIBPATH:C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\lib/x64" /LIBPATH:C:\DeepSjeng\train\venv10\libs /LIBPATH:C:\Python\Python36\libs /LIBPATH:C:\Python\Python36 /LIBPATH:C:\DeepSjeng\train\venv10\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\ATLMFC\lib\x64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.17763.0\um\x64" cudart.lib c10.lib caffe2.lib torch.lib torch_python.lib caffe2_gpu.lib _C.lib /EXPORT:PyInit_fused_adam_cuda build\temp.win-amd64-3.6\Release\apex/optimizers/csrc/fused_adam_cuda.obj build\temp.win-amd64-3.6\Release\apex/optimizers/csrc/fused_adam_cuda_kernel.obj /OUT:build\lib.win-amd64-3.6\fused_adam_cuda.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\apex/optimizers/csrc\fused_adam_cuda.cp36-win_amd64.lib
```

Maybe this is a bug in some torch header that does not set up exports/calling convention correctly?

**Comment by gcp** [comment]
Fixed in master + PyTorch 1.1.
